python/pdo/scripts/ConfigureCLI.py

```diff
@@ -0,0 +1,266 @@
+    config = parse_configuration_file(filename, node_map)
+
+    filename = os.path.join(options.output_directory, 'etc', node + '.toml')
+    with open(filename, 'w') as outfile:
+        toml.dump(config, outfile)
+
+    generate_service_keys(os.path.join(options.output_directory, 'keys', node))
+    add_service_to_site(service, node, config)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def configure_services() :
+    parser = argparse.ArgumentParser(description='Script to generate service configuration files from a template')
+
+    parser.add_argument(
+        '-c', '--count',
+        help='Number of service instances to configure in the format eservice, sservice, pservice',
+        default=[5, 5, 5], nargs=3, type=int)
+
+    parser.add_argument(
+        '-o', '--output-directory',
+        help='Name of the directory where generated configuration files are written',
+        default='etc')
+
+    parser.add_argument(
+        '-t', '--template-directory',
+        help='Directory in which the template configuration will be found',
+        default='etc/templates')
+
+    parser.add_argument(
+        '--set',
+        help='Specify arbitrary configuration options',
+        nargs=2, action='append')
+
+    options = parser.parse_args()
+
+    global config_map
+    config_map = build_configuration_map()
+
+    if options.set :
+        for (k, v) in options.set : config_map[k] = v
+
+    # Set up the directories
+    if not os.path.exists(options.output_directory) :
+        os.makedirs(options.output_directory)
+    if not os.path.exists(os.path.join(options.output_directory,'etc')) :
+        os.makedirs(os.path.join(options.output_directory,'etc'))
+    if not os.path.exists(os.path.join(options.output_directory,'keys')) :
+        os.makedirs(os.path.join(options.output_directory,'keys'))
+
+    # Generate EService configuration files and keys
+    for n in range(1, options.count[0]+1) :
+        expand_service(options, 'eservice', 'EnclaveService', n)
+
+    # Generate SService configuration files and keys
+    for n in range(1, options.count[1]+1) :
+        expand_service(options, 'sservice', 'StorageService', n)
+
+    # Generate PService configuration files and keys
+    for n in range(1, options.count[2]+1) :
+        expand_service(options, 'pservice', 'ProvisioningService', n)
+
+    # Generate enclave configuration file
+    expand_helper(options, 'enclave.toml')
+
+    filename = os.path.join(options.output_directory, 'etc', 'site.toml')
+    with open(filename, 'w') as outfile:
+        toml.dump(site_information, outfile)
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def configure_users() :
+    parser = argparse.ArgumentParser(description='Script to generate configuration files from a template')
```

Comment:
configure_users seem very different from configure_services. In a real deployment, I would imagine that we would never configure users as described here, specifically, we would expect users to come with their own keys. this method I guess is still for test purposes, and not intended beyond that for a real deployment.

Reply:
it is supposed to be very different. configure_services would be called to configure services, configure_users would be called to generate pcontract.toml mostly. the default for keys is 0 (that is... bring your own). you can make more if you want, but the default creates none.